pkg/deviceplugin/nic.go

```diff
@@ -3,69 +3,97 @@
 package deviceplugin
 
 import (
+	"errors"
 	"os"
-	"os/exec"
-	"regexp"
+	"path"
+	"strconv"
 	"strings"
 
 	"github.com/golang/glog"
 )
 
-// Takes the output from lshw and returns the device name for each solarflare
-// device.
-func parseOutput(output string) []string {
-	// "lshw -short -class network" sample output:
-	// H/W path            Device     Class          Description
-	// =========================================================
-	// /0/100/1b/0         enp2s0f0   network        XtremeScale SFC9250 10/25/40/50/100G Ethernet Controller
-	// /0/100/1b/0.1       enp2s0f1   network        XtremeScale SFC9250 10/25/40/50/100G Ethernet Controller
-	// /0/100/1c.1/0       eno1       network        NetXtreme BCM5720 Gigabit Ethernet PCIe
-	// /0/100/1c.1/0.1     eno2       network        NetXtreme BCM5720 Gigabit Ethernet PCIe
+const (
+	sysClassNetPath  = "/sys/class/net/"
+	solarflareVendor = "0x1924"
+)
+
+func isSFCNic(devicePath string) bool {
```

Comment:
Would it be worth using isOnloadableNic or isSupportedNic?

Just in case we are asked to support another vendor sometime in the future.